src/toolkits/object_detection/object_detector.hpp

```diff
@@ -109,6 +138,22 @@ class EXPORT object_detector: public ml_model_base {
   END_CLASS_MEMBER_REGISTRATION
 
  protected:
+  // Constructor allowing tests to set the initial state of this class and to
+  // inject dependencies.
+  object_detector(
```

Comment:
Nice! We should take motivation from this for other toolkits' C++ Tests as well.